### ISSUE TITLE
Expose HTML target attr to accessibility APIs

### ISSUE THREAD

**Issue by yatil** [w3c/html-aam#434]
[As outlined in the ARIA repository](https://github.com/w3c/aria/issues/1785), having an ARIA property that allows assistive technologies to announce or inform their users on the target of a link is useful and would also remove the need for authors to specify the text “opens in new window or tab” when they use `target="_blank"` on their websites.

The idea is that this attribute is specially set from the HTML value and that user agents then can add additional information that helps users to know what happens once they activate a link.

To quote from the linked ARIA issue: 

> `a@target="_blank"` could be matched to the `aria-link-target` property value of `new`. The browser should then amend this for assistive technologies to announce the setting that the user prefers for opening new contexts: “new tab” or “new window”.

**Comment by cookiecrook** [comment]
This could/should be mapped to Accessibility APIs without the need for a new ARIA attribute. E.g. WebKit could expose any use of the `target` attr to a property, not necessarily scope-locked to link or target. Maybe "AXPopupBehaviorSomethingOrOther"... The new ARIA attribute proposal would be useful in those authoring cases that ignore the first rule of ARIA: ~"Don't use ARIA if a native equivalent exists."

**Comment by cookiecrook** [comment]
I'd suggest changing the title to "Expose HTML target attr to accessibility APIs"

**Comment by yatil** [comment]
I think this aria attribute could also used to expose the download attribute for something screen readers could announce. That was one reason why I thought decoupling it from the Target attribute directly w/could be a good idea.

**Comment by cookiecrook** [comment]
Yes, that case may be more difficult to detect automatically by the browser since it may require pre-fetching the HTTP headers for the downloadable file.

[Update: Clarifying that some links point to resources that will download once you request them, but there is nothing indicating it until the HTTP GET goes through. So if we expose `<a href="#" download>` differently from `<a href="some-file-path-that-will-download">`, it could be result in an even more unpredictable user experience than currently.]

**Comment by sivakusayan** [comment]
In Blink, this is currently exposed in both IA2 and AT-SPI via the `link-target` attribute. [Here is the change](https://chromium-review.googlesource.com/c/chromium/src/+/4508963). I don't think Blink currently maps this on Mac, at least I couldn't find anything via a "search by reference" on an enum.

**Comment by cookiecrook** [comment]
Mac AX API property is AXURL

**Comment by medeaoblongata** [comment]
For good measure, I would like to mention that as well as `target`, `mailto:`, `tel:`, and `sms:`, urls should also have special representation in the accessibility tree, plus <strike>ping and</strike> `download`.

**Comment by scottaohara** [comment]
`mailto`, `tel` and `sms` are all arguably things that could / should be clear from the accessible name / context of links.... and if the're not clear, then it's likely unclear to everyone what would happen when activating.

One could make a similar argument about `download`, as often such instances would have some visual cue, if not the explicit word of 'download' as part of the link text / accName.  But, for instances where the text is ambigious to everyone, there's an argument that the act of downloading by the browsers can be more visually overt - and it would be nice to have that parity exposed programmatically.

I am unclear as to how `ping` comes into play here.  That's not an attribute that is presented to any user... so what's the rational for mentioning that attribute?

**Comment by medeaoblongata** [comment]
It's true that `mailto`, `tel` and `sms` *should* be clear from name / context, although I can easily imagine situations where `tel` and `sms` are difficult to distinguish. I am especially thinking that there is real potential for disorientation if a link named "contact us" opens another app (or tab) without due warning. I've certainly experienced accessibility audits that insisted on "opens in new tab/window" being appended to the link name, even though I have now convinced myself that there is no actual WCAG violation without it.

I think my main argument comes from the notion of "accessibility by default", that if these settings are included in the a11y tree, then the various AT user communities can more readily decide whether/how it is most useful to expose them, (**especially** in multilingual contexts). If they're not in the tree, that avenue is closed off entirely, and hacking the acc name (with attendant i18n considerations) is all that remains. 

IMO, hacking the name feels "dirty", and I dare say the extra affordance is liable to be skipped, overlooked or implemented inconsistently. I would argue that if user agents already surface these attributes visually, which they do (e.g. in the status line, or with a change of cursor on hover), then they should **also** be exposed in the a11y tree.

By my reading, "Info and Relationships" is met if `download` and `target` are programmatically determined, But then if nothing useful is actually done with them (by ATs), this seems like a wasted opportunity for real improvement: Furthermore, authors can be "letter-of-the-law" compliant without actually providing the intended benefit of a clear affordance.

You are right that ping is more of a backend thing, so I will strike that out in the post above.

**Comment by yatil** [comment]
As much as I think revealing related information might be useful for users, this issue was about a very specific use case. It might be useful to open a new issue with other ideas.
